sql/core/src/main/scala/org/apache/spark/sql/internal/SharedState.scala

```diff
@@ -69,6 +71,29 @@ private[sql] class SharedState(val sparkContext: SparkContext) extends Logging {
   val jarClassLoader = new NonClosableMutableURLClassLoader(
     org.apache.spark.util.Utils.getContextOrSparkClassLoader)
 
+  /**
+   * Add a global-scoped jar
+   */
+  def addJar(path: String): Unit = {
+    if (sparkContext.conf.get(CATALOG_IMPLEMENTATION) == "hive") {
+      // Hive metastore supports custom serde.
+      externalCatalog.addJar(path)
```

Comment:
We call `addJar` only when the external catalog is `Hive metastore`. This is just consistent with our current implementation: 

https://github.com/apache/spark/blob/261c55dd8808502fb7f3384eb537d26a4a8123d7/sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveSessionState.scala#L104